lib/langchain/vectorsearch/elasticsearch.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+module Langchain::Vectorsearch
+  class Elasticsearch < Base
+    attr_accessor :es_client, :index_name, :options
+
+    def initialize(url:, index_name:, llm:, api_key: nil, es_options: {})
+      require "elasticsearch"
+
+      @options = {
+        url: url,
+        request_timeout: 20,
+        log: false
+      }.merge(es_options)
+
+      @es_client = ::Elasticsearch::Client.new(**options)
+      @index_name = index_name
+
+      super(llm: llm)
+    end
+
+    def add_texts(texts: [])
+      body = texts.map do |text|
+        [
+          {index: {_index: index_name}},
+          {input: text, input_vector: llm.embed(text: text).embedding}
+        ]
+      end.flatten
+
+      es_client.bulk(body: body)
+    end
+
+    def update_texts(texts: [], ids: [])
+      body = texts.map.with_index do |text, i|
+        [
+          {index: {_index: index_name, _id: ids[i]}},
+          {input: text, input_vector: llm.embed(text: text).embedding}
+        ]
+      end.flatten
+
+      es_client.bulk(body: body)
+    end
+
+    def create_default_schema
+      es_client.indices.create(
+        index: index_name,
+        body: default_schema
+      )
+    end
+
+    def delete_default_schema
+      es_client.indices.delete(
+        index: index_name
+      )
+    end
+
+    def default_vector_settings
+      {type: "dense_vector", dims: 384}
```

Comment:
Where does `dims: 384` come from? 

You can see other vectorsearch providers, we all use [this method](https://github.com/andreibondarev/langchainrb/blob/main/lib/langchain/llm/base.rb#L25-L27) to determine the `dimension` metric. If there's a mismatch -- the querying won't work correctly.

Reply:
I pushed up a fix.